src/azul/service/repository_controller.py

```diff
@@ -224,3 +237,13 @@ def download_file(self,
             }
         else:
             assert False
+
+    def accessible_sources(self,
+                           catalog: CatalogName,
+                           authentication: Authentication,
+                           ) -> JSONs:
+        plugin = self.repository_plugin(catalog)
+        lister = plugin.source_lister(authentication)
+        sources = self.source_cache_service.list_sources(lister)
+        catalog_sources = {source.name for source in plugin.sources}
+        return [source for source in sources if source['sourceName'] in catalog_sources]
```

Comment:
@hannes-ucsc : "Business logic of filtering by catalog belongs in RepositoryService"